solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -1286,12 +1286,12 @@ public void inform(SolrCore core) {
       boolean backupOnCommit = backup.contains("commit");
       boolean backupOnOptimize = !backupOnCommit && backup.contains("optimize");
       List<?> replicateAfter = leader.getAll(REPLICATE_AFTER);
-      replicateOnCommit = replicateAfter.contains("commit");
+        replicateOnCommit = replicateAfter.contains("commit");
       replicateOnOptimize = !replicateOnCommit && replicateAfter.contains("optimize");
 
-      if (!replicateOnCommit && !replicateOnOptimize) {
+      /*if (!replicateOnCommit && !replicateOnOptimize) {
         replicateOnCommit = true;
-      }
+      }*/
```

Comment:
what is the significance of this change, wrt the overall goal of the PR?

Reply:
Looks like ReplicationHandler registers to updateHandler. We may need to add another api to just register the handler in updateHandler(don't open searcher for it - that above getUpdateHandler change).
```
public UpdateHandler getUpdateHandler() {
    // hitesh
    if (!isVersionBucketSeeded.get()) {
      synchronized (isVersionBucketSeeded) {
        seedVersionBuckets();
        isVersionBucketSeeded.set(true);
      }
    }
```
Basically two versions of this api or proper code organization.